fpm/src/fpm_environment.f90

```diff
@@ -51,4 +52,16 @@ integer function get_os_type() result(r)
         r = OS_LINUX
     end if
     end function
-end module
+
+    subroutine run(cmd)
+        character(len=*), intent(in) :: cmd
+        integer :: stat
+        print *, "+ ", cmd
+        call execute_command_line(cmd, exitstat=stat)
+        if (stat /= 0) then
+            print *, "Command failed"
+            error stop
```

Comment:
Should have an exit code (see #150 and #143). It would probably be better to avoid `error stop` invocations everywhere other than the top-level module or the program unit.

Reply:
On my machine (Ubuntu) Fortran `error stop` already propagates a non-zero exit code to the calling shell.

> It would probably be better to avoid error stop invocations everywhere other than the top-level module or the program unit.

Agreed - would you recommend propagating up with `error_t` and catching in `fpm.f90` as is already done with `get_package_data`? Happy to do so, but I'm wary of the size of this PR already so probably a job for a separate PR.

Reply:
My error, this was regarding bootstrap fpm.  Everything fine with just `error stop` than.

Proper error handling can be a separate PR, indeed.